File: src/ex_20_OOPs_Polymorphism/methodoveridding/Lab190_Methodoverrding_Automation.java
package ex_20_OOPs_Polymorphism.methodoveridding;

public class Lab190_Methodoverrding_Automation {
    public static void main(String[] args) {
        ChromeTC c1 = new ChromeTC();
        c1.openBrowser();

        FirefoxTc f1 = new FirefoxTc();
        f1.openBrowser();

        CommonToAll commonToAll = new CommonToAll();
        commonToAll.openBrowser();

        // Dynamic Dispatch
        CommonToAll c2 = new ChromeTC();
        c2.openBrowser();

        CommonToAll c3 = new FirefoxTc();
        c3.openBrowser();

//        ChromeTC cc = new FirefoxTc();





    }
}


class CommonToAll{
    void openBrowser(){
        System.out.println("Starting the IE browser.");
    }
}

class ChromeTC extends CommonToAll{
    @Override
    void openBrowser() {
        System.out.println("Starrting Chrome, Better Browser!!");
    }
}

class FirefoxTc extends CommonToAll{
    @Override
    void openBrowser() {
        System.out.println("Starrting FirefoxTc, Better Browser!!");
    }
}

File: javascript/ex_20_OOPs_Polymorphism/methodoveridding/Lab190_Methodoverrding_Automation.js
// Method Overriding - Automation Example

class CommonToAll {
    openBrowser() {
        console.log("Starting the IE browser.");
    }
}

class ChromeTC extends CommonToAll {
    openBrowser() {
        console.log("Starrting Chrome, Better Browser!!");
    }
}

class FirefoxTc extends CommonToAll {
    openBrowser() {
        console.log("Starrting FirefoxTc, Better Browser!!");
    }
}

const c1 = new ChromeTC();
c1.openBrowser();

const f1 = new FirefoxTc();
f1.openBrowser();

const commonToAll = new CommonToAll();
commonToAll.openBrowser();

// Dynamic Dispatch
const c2 = new ChromeTC();
c2.openBrowser();

const c3 = new FirefoxTc();
c3.openBrowser();

// const cc = new FirefoxTc(); // ChromeTC cc = new FirefoxTc(); -- not valid in Java either
